### PR TITLE
Refactor most least accurate teams

### PR DESCRIPTION
**Pull request by s-steel**
Refactor most and least accurate team methods.
Add new helper methods and create tests for them.
All tests passing.

### CODE REVIEW

File: lib/season_stats_module.rb [hunk 1]
```diff
@@ -39,33 +39,43 @@ def worst_coach(season_id)
...
-  def most_accurate_team(season_id)
-    team_hash = Hash.new
-    season_teams(season_id).each do |team|
-      total_shots = game_teams_data_for_season(season_id).sum do |game|
-        if game.team_id == team
-          game.shots
-        else
-          0
-        end
+  def total_shots_by_team(season_id, team)
```
Comment (c-bartell) [inline lib/season_stats_module.rb:42]: Switching this from a hash to an array was a good call. It makes it easier to follow.

File: lib/season_stats_module.rb [hunk 2]
```diff
@@ -39,33 +39,43 @@ def worst_coach(season_id)
...
+  end
 
-      total_goals = game_teams_data_for_season(season_id).sum do |game|
-        if game.team_id == team
-          game.goals
-        else
-          0
-        end
+  def total_goals_by_team(season_id, team)
+    total_goals = game_teams_data_for_season(season_id).sum do |game|
```
Comment (c-bartell) [inline lib/season_stats_module.rb]: I think you could take out the local variable and just return the value of the sum.

**Review by c-bartell — CHANGES_REQUESTED** [review]
I think that once the merge conflicts and the local variable in total_goals_by_team are taken care of, this will be good to merge.

File: lib/season_stats_module.rb (hunk 2 above)
Comment (s-steel) [reply lib/season_stats_module.rb]: Nice eye, it has been removed for this method and the total shots method.

**Review by c-bartell — APPROVED** [review]

**Pull request merged**